src/CMakeLists.txt

```diff
@@ -678,9 +683,9 @@ add_executable(ceph-osd ${ceph_osd_srcs}
   $<TARGET_OBJECTS:common_util_obj>)
 add_dependencies(ceph-osd erasure_code_plugins)
 target_link_libraries(ceph-osd osd os global ${BLKID_LIBRARIES})
-if(WITH_FUSE)
+if(HAVE_LIBFUSE)
```

Comment:
@tchaikov 
If you look at the code in ./CMakeLists.txt the FUSE testing actually purposly sets HAVE_LIBFUSE in the case that FUSE is actually present.
So IMHO it is better to actually use `HAVE_LIBFUSE`, because that guards the fact that we did not set `-DWITH_FUSE=OFF` but still do not have FUSE available.

Note: I agree that having both (and using) `WITH-*` and `HAVE_*` does not help.
Perhaps it should be that the `WITH_*` switches are for commandline-CMake, and should resulting in a matching `HAVE_*`